numpy/core/function_base.py

```diff
@@ -1,11 +1,25 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
 
+def index_deprecate(i):
+    try:
+        i = operator.index(i)
+    except TypeError:
+        i = int(i)
+        msg = "Indexing with a float is deprecated "\
+        "and will be removed in the future."
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
```

Comment:
MIght be worth making it an input parameter to the function with a default of `3`. There may be functions that want a higher level.